app/domain/authentication/authn_jwt/validate_status.rb

```diff
@@ -134,25 +144,15 @@ def webservice
       end
 
       def validate_signing_key
-        fetch_signing_key.call
-        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
-      end
-
-      def fetch_signing_key
-        @fetch_signing_key ||= ::Util::ConcurrencyLimitedCache.new(
-          ::Util::RateLimitedCache.new(
-            fetch_signing_key_interface,
-            refreshes_per_interval: ::Authentication::AuthnJwt::CACHE_REFRESHES_PER_INTERVAL,
-            rate_limit_interval: ::Authentication::AuthnJwt::CACHE_RATE_LIMIT_INTERVAL,
-            logger: @logger
-          ),
-          max_concurrent_requests: ::Authentication::AuthnJwt::CACHE_MAX_CONCURRENT_REQUESTS,
-          logger: @logger
+        @fetch_signing_key.call(
+          cache_key: signing_key_interface.signing_key_uri,
+          signing_key_interface: signing_key_interface
         )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
       end
-
-      def fetch_signing_key_interface
-        @fetch_signing_key_interface ||= @create_signing_key_interface.call(
+      
+      def signing_key_interface
```

Comment:
Is it aligned in behaviour with the next remark from the main PR
https://github.com/cyberark/conjur/blob/c4eedc7b95ee1fb5bd5a9581300614f13bbe880c/app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb#L46

Reply:
@sashaCher Class is created per request so the members under the class like `@signing_key_interface` are not static.
What is static is the dependencies of the command class which there the cache is defined.

@tzheleznyak Do you agree ?

Reply:
lets talk about it